src/screenshots/crop.js

```diff
@@ -112,22 +109,33 @@ function copyImagePart (srcImage, { left, top, width, height }) {
     return dstImage;
 }
 
-export default async function (screenshotPath, markSeed, clientAreaDimensions, cropDimensions) {
-    const srcImage  = await readPng(screenshotPath);
+export async function cropScreenshotBinary (path, markSeed, clientAreaDimensions, cropDimensions, binaryImage) {
+    let png = await readPng(binaryImage);
+
+    const clippingArea = detectClippingArea(png, markSeed, clientAreaDimensions, cropDimensions);
 
-    const clippingArea = detectClippingArea(srcImage, { markSeed, clientAreaDimensions, cropDimensions, screenshotPath });
+    if (!clippingArea)
+        throw new Error(renderTemplate(WARNING_MESSAGES.screenshotMarkNotFound, path, markSeedToId(markSeed)));
 
-    if (clippingArea.width <= 0 || clippingArea.height <= 0) {
-        await deleteFile(screenshotPath);
+    if (clippingArea.width <= 0 || clippingArea.height <= 0)
         throw new InvalidElementScreenshotDimensionsError(clippingArea.width, clippingArea.height);
-    }
 
-    if (!markSeed && !cropDimensions)
-        return true;
+    if (markSeed || cropDimensions) {
+        png = copyImagePart(png, clippingArea);
+
+        await writePng(path, png);
```

Comment:
I think it's better to extract all file-related logic to a different place and keep `cropScreenshotBinary` for image operations only.